### DISCUSSION TITLE
userprefs.config

### DISCUSSION THREAD

**Discussion by se7en-x230** [prasanthrangan/hyprdots#765] [General]
Does the userpref.conf overwrite settings?
Meaning do I have to unbind keys in this file before I assign a new key or is it enough to just add my preferences.
Also can I set the default apps in there if I would rather use another browser?
Can configs made to waybar be conserved or are they overwritten by ./install.sh -r

**Comment by kRHYME7** [comment]
Hello, normally keybindings.conf is now user preference so you can change your keybinding there now.(won't be overwritten if the file already exists)

 But, yes you need to unbind it first. 
 check ./restore_cfg.lst the first column is to be restored and 2nd column will be backed up.

**Comment by se7en-x230** [comment]
Thanks I understand 
so as keybindings.conf does not get touched when updating adding 
$term = st
to userprefs.config is not necessary as I understand.

But I guess using the userprefs.conf is good practice as in case of a new install all I would need to copy over would be this file.

Will the update overwrite my waybar modules choice/edits?

**Reply by kRHYME7** [reply]
Check this out 

```
❯ cat ~/Hyprdots/Scripts/restore_cfg.lst
N|Y|${HOME}|.zshrc .p10k.zsh|zsh oh-my-zsh-git zsh-theme-powerlevel10k pokemon-colorscripts-git
Y|Y|${HOME}/.config/Code/User|settings.json|visual-studio-code-bin
Y|Y|${HOME}/.config|dunst|dunst
Y|Y|${HOME}/.config/hypr|animations.conf hyprland.conf theme.ctl windowrules.conf scripts themes wallbash|hyprland
N|Y|${HOME}/.config/hypr|keybindings.conf monitors.conf userprefs.conf|hyprland
N|Y|${HOME}/.config/hypr|nvidia.conf|hyprland nvidia-utils
Y|Y|${HOME}/.config|kitty|kitty
Y|Y|${HOME}/.config|Kvantum|kvantum
Y|Y|${HOME}/.config|MangoHud|mangohud
Y|Y|${HOME}/.config|neofetch|neofetch
Y|Y|${HOME}/.config|qt5ct|qt5ct
Y|Y|${HOME}/.config|rofi|rofi
Y|Y|${HOME}/.config|swaylock|swaylock-effects
Y|N|${HOME}/.config|swww|swww
Y|Y|${HOME}/.config|waybar|waybar
Y|Y|${HOME}/.config|wlogout|wlogout
Y|Y|${HOME}/.config|nwg-look gtk-3.0 xsettingsd|nwg-look
Y|Y|${HOME}/.icons|default|nwg-look
Y|Y|${HOME}|.gtkrc-2.0|nwg-look
Y|Y|${HOME}/.config|dolphinrc kdeglobals|dolphin
Y|Y|${HOME}/.local/share|dolphin|dolphin
Y|Y|${HOME}/.local/share/kxmlgui5|dolphin|dolphin


```


you can modify that so that restoring and backing up configurations can be managed. 

How I wish someone could document this on the wiki.
